src/code.lisp

```diff
@@ -4,24 +4,29 @@
 (defclass postgres-query (vellum.header:frame-range-mixin
                           cl-ds:traversable)
   ((%query :initarg :query
-           :reader read-query)))
+           :reader read-query)
+   (%parameters :initarg :parameters
+                :reader read-parameters)
+   (%query-name :initarg :query-name
+                :reader read-query-name)))
 
 
-(defun postgres-query (query header)
-  (make 'postgres-query :query query :header header))
+(defun postgres-query (query header &optional parameters query-name)
```

Comment:
Is there merit in having query-name without parameters? If yes, reorder arguments, query-name should go first in the lambda-list

Reply:
hmm...actually, I don't think this is the best idea...

Maybe it'd be better if we add a separate case for prepared queries

```lisp
(defclass postgres-prepared-query (postgres-query))

(defvar *query* (vellum:prepare-query ...))

(vellum:copy-from :postmodern
                              *query*
                              :parameters (list ...)
                              :columns (list ...))
```

...and when we actually query, we dispatch on the class. Otherwise, it would

Reply:
That is also an option.

Reply:
Done. I left `defprepared`, for now, because it will change how it iterates over the results. I can do it after this one.